### ISSUE TITLE
[Mod Conflict] Dimensional Dungeons prevents zombie and skeleton horses from burning in daylight

### ISSUE THREAD

**Issue by NJL5390** [AllenSeitz/DimDungeons#26]
Dimensional Dungeons version: 1.093
Zombie horse spawn version: 2.7
Collective version: 2.25
Skeleton horse spawn version: 1.6
Forge Version: 36.1.2
Minecraft Version: 1.16.5
Latest Log: https://pastebin.com/areDpi7H

I've been using some mods recently that allow skeleton and zombie horses to spawn naturally throughout the world on the surface at night. One of the features of these mods is that it causes zombie and skeleton horses to burn in sunlight to prevent them from overpopulating the world over the course of several nights.

For some reason Dimensional Dungeons seems to prevent this from happening and I have no idea why.

Without Dimensional Dungeons:
![2021-03-25_16 20 14](https://user-images.githubusercontent.com/64381954/112509137-1908ae00-8d88-11eb-8afb-b2da62eab4a2.png)

With Dimensional Dungeons: 
![2021-03-25_16 09 16](https://user-images.githubusercontent.com/64381954/112509322-45242f00-8d88-11eb-86dd-666035c311b4.png)

Here's links to the mod pages: 
https://www.curseforge.com/minecraft/mc-mods/skeleton-horse-spawn
https://www.curseforge.com/minecraft/mc-mods/zombie-horse-spawn

**Comment by NJL5390** [comment]
I've opened an issue about on their github too: https://github.com/ricksouth/serilum-mc-mods/issues/519

**Comment by AllenSeitz** [comment]
Thank you for the detailed report including a log, the versions of all mods and minecraft itself, and screenshots!

Unfortunately I have no idea, either. That is such a weird, tangential thing, I just can't imagine what would cause that. I do cancel a number of forge events (block break, on right click, etc) as part of my block protection implementation. But in all of that code I put a big fat "if ( event.dimension != dimdungeons:dungeon_dimension) then abort" at the top.

I will look into this because I am curious. But I'm also trying to get 1.1 out the door 4/1 and I want to push a beta of 1.1 maybe today or tomorrow.

Actually one guess. Maybe see if installing any dimension mod or datapack causes this? Maybe the mobs are hardcoded to burn in "dim 0" or something and re-indexing the dimensions breaks it? In that case this is a vanilla bug that Mojang would've overlooked because they consider the mobs unused.

**Comment by NJL5390** [comment]
> Actually one guess. Maybe see if installing any dimension mod or datapack causes this? Maybe the mobs are hardcoded to burn in "dim 0" or something and re-indexing the dimensions breaks it? In that case this is a vanilla bug that Mojang would've overlooked because they consider the mobs unused.

Hmm, Interesting, I'll look into that when I next have the time.

> I will look into this because I am curious. But I'm also trying to get 1.1 out the door 4/1 and I want to push a beta of 1.1 maybe today or tomorrow.

In the meantime, don't worry too much about it, I can wait for a fix if need be, I'm rather busy these days anyway. I look forward to seeing what content you have planned for the future!

**Comment by NJL5390** [comment]
> Actually one guess. Maybe see if installing any dimension mod or datapack causes this? Maybe the mobs are hardcoded to burn in "dim 0" or something and re-indexing the dimensions breaks it? In that case this is a vanilla bug that Mojang would've overlooked because they consider the mobs unused.

After some testing, This guess seems to be correct. The zombie and skeleton horses don't burn if I have a dimension mod installed (i.e a mod that adds a new dimension). I tested with The Undergarden mod and The Bumblezone Mod and the same thing happened, the zombie and skeleton horses refused to burn in daylight.

Therefore, it would seem that this issue has nothing to do with your mod as it seems to happen with every dimension mod I installed, I just happened to find it with yours because it was the only dimension mod I had installed.
